### PR TITLE
rubocop: Enable RSpec/RepeatedExampleGroupDescription cop & fix offenses

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

- Inspired by my other RuboCop PR on Friday, here's another TODO dealt with.

### CODE REVIEW

File: Library/Homebrew/test/download_strategies/curl_spec.rb [hunk 1]
```diff
@@ -147,22 +125,48 @@
...
+          HOMEBREW_CACHE/"downloads/1ab61269ba52c83994510b1e28dd04167a2f2e8393a35a9c50c1f7d33fd8f619--foo.tar.gz",
+        )
+      }
+    end
+
     context "with a file name trailing the URL path" do
       let(:url) { "https://example.com/cask.dmg" }
 
-      its("cached_location.extname") { is_expected.to eq(".dmg") }
+      it {
```
Comment (Rylan12) [inline Library/Homebrew/test/download_strategies/curl_spec.rb:151]:
```suggestion
      it "returns a file with a `.dmg` extension" {
```

Might be a good idea to add some sort of description like this to the `its` to `it` conversions.

Without a description:
```
CurlDownloadStrategy#cached_location with a file name trailing the URL path is expected to eq ".dmg"
```

With a description:
```
CurlDownloadStrategy#cached_location with a file name trailing the URL path returns a file with a `.dmg` extension
```

**Review by Rylan12 — APPROVED** [review]
Looks good. One suggestion that isn't crucial

File: Library/Homebrew/test/download_strategies/curl_spec.rb (hunk 1 above)
Comment (issyl0) [reply Library/Homebrew/test/download_strategies/curl_spec.rb:151]: I mostly didn't add descriptions as the other `it {`s that I moved down didn't have any. 🤷🏻

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Nice work @issyl0!

**Pull request merged**